### ISSUE TITLE
Problem Turning Pages under iOS 8 on iPad 3

### ISSUE THREAD

**Issue by Faterson** [kguil/Marvin-Roadmap#362]
Ever since upgrading to iOS 8 (and what a nightmare it was, generally speaking...) and the current Marvin version on all of my eligible iDevices, I'm seeing issues with turning pages on iPad 3. Newer iDevices are performing fine. But in Marvin on iPad 3, every book is problematic, every chapter, regardless of its length. I can make a short YouTube video if that might help. Specifically, whenever I turn a page on iPad 3, it's not only slow, but also one half of the screen, or one third of the screen, appears completely black for a split-second. It almost reminds you of the "black flash" of Kindle e-ink devices, except that the "black flash" in Marvin "only" covers one half or one third of the screen when turning a page. This happens under all page-turning modes (I tried my default "None", as well as "Fade" and "Slide", but the "black flash" occurs everywhere).

**Comment by Marvinlover1** [comment]
It's reasons like this that has forced me to budget to purchase a new iPad this spring. But my trusty iPad 3 is long in the tooth and is definitely in need of upgrading anyway. My only concern is if I budget for the newer 12 inch version coming out...will Marvin be designed to read at that screen size? Or will I have to budget for the normal 8 inch version?

**Comment by Faterson** [comment]
The "normal" version is the 10-inch iPad, not 8-inch (which would be iPad mini). I would definitely recommend the iPad Air 2 to anyone. I don't possess it myself, but I've seen it in action. It's the first iPad with 2 GB of RAM, instead of 1 GB as the prevous iPad Air or the minis (or all the iPhones!), and it shows.

I'd definitely be interested in a 12-inch or 13-inch iPad, but as far as I know, that is nothing more than just a rumour at this point. It may never be released.

I have more information on the bug now... The "black flash" only occurs in night mode because... well, it's night-time here right now (4 hours past midnight of New Year's Eve), so I had night mode active in my Marvin. If I switch to a day mode, the issue is still there, but it's no longer a "black flash".

Here is what's happening: when you turn a page on iPad 3 now, Marvin fails to load the *full* page content for a spilt second. So, instead, one half of the screen, or one third of the screen (this changes irregularly from page to page), appears empty -- only showing your selected reading background color. Only later on, the full text appears and fills the entire screen. (In night mode, this then creates a "black flash" effect, due to the night mode employing a dark, typically black, background color.) 

I can make that short YouTube clip to document the issue, if that helps.

**Comment by Marvinlover1** [comment]
Thanks for the info on the memory specs. Yes, 2 gigs of RAM is alot better than 1 gig of RAM. As for the iPad size...I think what I'm concerned about is if I purchase the "bigger" Ipad version (assuming Apple does release it) my concern will be whether or not Marvin will need to be updated to take advantage of that screen size...otherwise it might not be worth getting the larger new iPad and stick with the iPad Air 2.

**Comment by Faterson** [comment]
I'm no software developer, but I don't think that will be a issue. If Marvin for iPhone can support all 4 iPhone sizes currently (from 3.5 inches all the way to 5.5 inches), then supporting 3 iPad sizes from 8 to 12 or 13 inches shouldn't be a problem. (Especially for reading scholarly e-books, I really might prefer a large iPad, whereas for novels or short stories, iPad mini appears ideal.)

**Comment by Marvinlover1** [comment]
But the issue might be one of pixel density used assuming Apple decides to release a larger size iPad. That's the real issue because if pixel density is increased for that screen size...then Marvin would have to be tweaked to take advantage of that size. I wouldn't get that iPad version unless I hear definitively that Kris will release a new version of Marvin that takes advantage of that screen size.

**Comment by Faterson** [comment]
Right... but just as Kris tweaked Marvin for iPhone 6 Plus's higher pixel density compared to the iPhone 6, 5S, etc. (specifically, 401 vs. 326 ppi), in the same way Marvin would be tweaked for a larger iPad compared to the current iPad sizes. I know that's easier said than done, and it might cause countless sleepless nights for developers, but I have zero doubt that it *would* be done.  :-)

**Comment by Marvinlover1** [comment]
I can't believe I'm saying this...but my decision to get the iPad Air 2 vs an IPad Air King (12 inch plus) is almost entirely dependent on whether Kris will make a version of Marvin optimized for the iPad Air King.

**Comment by Faterson** [comment]
Marvin would be among my top criteria, too, but also simply for browsing webpages, extra screen real estate is always good to have. Lots of content gets crammed into a typical webpage nowadays. 

(This can be particularly problematic for smaller Android tablets with their not-so-optimal 9:16 aspect ratio; iPad mini gets away with a smaller screen size thanks to the more comfortable 3:4 aspect ratio. I do prefer watching movies on my Android tablet thanks to its 16:9 aspect ratio, but for everything else, especially browsing the web, using a smaller 9:16 Android tablet in portrait mode tends to be painful.)

**Comment by Faterson** [comment]
I can now observe the same "black flash" issue on my iPad mini 2 and iPhone 5S, too, except that it's much less pronounced there. I probably wouldn't have noticed it on those 2 newer devices, if the bug wasn't so prominent on iPad 3. On the newer iPad and iPhone, the black flash is there, too, but it's very brief, and does not occur on every page, but maybe on every 4th or every 5th page. Whereas on iPad 3, the same bug is extremely noticeable, creating a very ugly effect upon every page turn.

None of this was occurring under iOS 7; I was using Marvin on a daily basis on all 3 of those devices until I upgraded to iOS 8 a couple of days ago. (Unwillingly; forced by the reckless practice of abandoning the support of earlier iOS versions almost as soon as a new iOS version comes out; Kris doesn't do this, but many developers do, even of apps I highly appreciate. Let me only say: Day One, Mailbox, iA Writer Pro...)

**Comment by Faterson** [comment]
Restarting iPad 3 doesn't resolve the issue; page-turning in Marvin is just as bad as before. Over in <a href="https://getsatisfaction.com/atracker/topics/icloud-syncing-not-really-usable-regularly-leads-to-data-corruption-data-loss?topic-reply-list%5Bsettings%5D%5Bfilter_by%5D=all&topic-reply-list%5Bsettings%5D%5Breply_id%5D=15181354#reply_15181354">the ATracker forum</a>, the developer is reporting that iOS 8's management of memory assigned to running apps is a lot worse in iOS 8 compared to iOS 7. Many apps are freezing and crashing for him, even on his new iPhone 6 Plus, that were having no problems under iOS 7. This is likely the problem for Marvin, too: iOS 8 is not assigning enough memory to Marvin for page-turning to proceed smoothly, for *textual* screen content to be loaded immediately upon every page turn, etc.

This morning, I was also unable to read the news on iPad 3, on Slovakia's top news website SME.sk. Yes, I know that that news site is badly designed, but the point is, I was always able to browse it normally under iOS 7. This morning, I couldn't, on iPad 3. Restarting iPad 3 didn't help, and trying all sorts of browsers didn't help, either: Mercury (my no. 1 preferred browser), Safari, Chrome, Dolphin, Atomic Web Browser, even the Chinese Maxthon... The website just kept freezing and crashing in every browser. On iPad mini 2, the same site could be browsed, albeit slowly. Other news sites, such as CNN.com, could be browsed even on iPad 3.

Thank you, Apple. This is what the recurring nightmare of annual iOS upgrades does to us. I hope someone at Apple will come to their senses, arranging it so that Apple will only start releasing new iOS versions every 2 years (or even less frequently), to give third-party apps developers some breathing room.

**Comment by Marvinlover1** [comment]
This is a very interesting thread. Thanks for posting this information. As of right now, I think the developer for Marvin has a problem and that's designing Marvin for the right OS. At this point I believe he should forgo development for iOS 8 and save the Marvin 3.0 release for iOS 9. The reason why I think he should take this direction is because this iOS upgrade will supposedly finally support multi tasking. For multi tasking to work properly...they'll need to redesign the OS to utilize memory more efficiently. And I suspect iOS 8 is failing because it's not really designed for 64 bit apps. This explains Apple's hard line about the February deadline to not allow non 64 bit updates to any app on the App Store. It also explains why Apple is bravely taking the stand to purge all non 64 bit apps on their Appstore in July. This means there will be a lot of apps that will fall by the wayside because they won't be 64 bit compliant. The gravy train for those developers is coming to an end unless they update their apps for the 64 bit environment. This is yet another reason why I'm taking the plunge this spring to purchase a new iPad. I'll wait to see the pricing, specs for this new version of the iPad and then make my decision. If I had to guess...I think the Marvin app won't be tweaked to take advantage of the new Ipad size when it comes out because this newer version is NOT slated to replace the current iPad Air edition but will complement it. This means it's more niche than mainstream. In this case...I'm leaning towards the iPad Air (whatever generation is current) and get that one.

**Comment by Faterson** [comment]
iOS 8 does look like a hopeless case. I *knew* it would be a nightmare to upgrade to it, and it is. But I was given no choice, given the panic with which third-party app developers were ditching iOS 7. I see lots of issues under iOS 8 in many apps – crucial for me – which performed flawlessly under iOS 7. Dropbox, DropVox, Day One... and now also Marvin.

I'm sorry to report this, but the "black flash" when turning pages is now also noticable on my iPad mini 2 (especially when *tapping*, rather than swiping, to turn a page), in the same book where it wasn't noticable a couple of days ago. I guess the iPad's memory must be exhausted, or something. I force-quit all running apps except Marvin, but it doesn't help. Perhaps restarting the whole iPad would help, but I couldn't be bothered. iOS 8 is just a big steaming heap of ****. I do appreciate its advanced functionality compared to iOS 7... but I thought it was Microsoft, not Apple, who were bringing advanced features at the expense of system stability and user experience. Perhaps, in the post-Jobs era, there is no longer any difference between Apple and Microsoft.
